common/views/components/TextInput/TextInput.js

```diff
@@ -1,66 +1,221 @@
 // @flow
-import { forwardRef, type ComponentType } from 'react';
+import { forwardRef, useContext } from 'react';
 import styled from 'styled-components';
-import Space, { type SpaceComponentProps } from '../styled/Space';
+import Icon from '../Icon/Icon';
+import { AppContext } from '@weco/common/views/components/AppContext/AppContext';
 import { classNames } from '../../../utils/classnames';
 
-const VisuallyHidden = styled.div`
-  border: 0;
-  clip: rect(0 0 0 0);
-  height: 1px;
-  margin: -1px;
+const TextInputWrap = styled.div.attrs({
```

Comment:
Should we be setting the props here?
(can't wait to move to typescript)

Reply:
props for styled components? I don't _think_ that's something we do?

Reply:
It was weirdly erroring in my IDE - why is it something we shouldn't start doing?

Reply:
Perhaps If its a styled component of a non built-in type, e.g. `const blah = styled(Space)`, maybe that would make sense (assuming Flow stops working if a component is wraped in `styled`).

But do you think it's required for built-in elements (listing all the attributes that e.g. a `div` _could_ accept)?

Reply:
It feels like having some type checking around custom props would be good. e.g. `props.big` - this  would catch a whole load of potential bugs we would have to write tests to protect against.

Using an inexact object type would work e.g.
```js
const TextInputWrap: ComponentType<{
  big: boolean,
  hasErrorBorder: boolean,
}> = styled.div.attrs({})
```

Reply:
I haven't dug much deeper. This is also bikeshedding of course.